### PR TITLE
[WIP] Try to force net core qualifiers to analyze Mojave build breaks

### PR DESCRIPTION
**Pull request by Kr1sso**
An attempt to fix build breaks on Azure VMs running on latest macOS Mojave. Feel free to ignore this.

### CODE REVIEW

File: Public/Src/Cache/VerticalStore/Deployment.dsc [hunk 1]
```diff
@@ -4,34 +4,22 @@
...
-};
-
+namespace Deployment {
+    @@public
+    export const deployment: Deployment.Definition = {
+        contents: [
+            InMemory.dll,
+            Interfaces.dll,
+            BasicFilesystem.dll,
+            BuildCacheAdapter.dll,
```
Comment (dannyvv) [inline Public/Src/Cache/VerticalStore/Deployment.dsc]: You might want to check with David & Lance: They were checking if we could remove these..

File: Public/Src/Deployment/buildXL.dsc [hunk 2]
```diff
@@ -21,9 +21,9 @@ namespace BuildXL {
             // analyzers
             importFrom("BuildXL.Tools").Execution.Analyzer.exe,
             importFrom("BuildXL.Tools").BxlScriptAnalyzer.exe,
-            ...(qualifier.targetFramework === "netcoreapp2.2" ? [] : [
+            ...addIf(qualifier.targetFramework !== "netcoreapp2.2",
```
Comment (dannyvv) [inline Public/Src/Deployment/buildXL.dsc]: >ifier.targetFramework !== "netcoreapp2.2 [](start = 25, length = 40)

If we will never support this on .net core (i.e. not even on 3.0 we should use `addIf(BuildXLSdk.isFullFramework,`

File: Public/Src/Cache/VerticalStore/Deployment.dsc (hunk 1 above)
Comment (Kr1sso) [reply Public/Src/Cache/VerticalStore/Deployment.dsc]: This is semantically still the same, just without duplication. I'll let them decide once they refactor.

File: Public/Src/Deployment/buildXL.dsc (hunk 2 above)
Comment (Kr1sso) [reply Public/Src/Deployment/buildXL.dsc]: Speaking of, I don't know a single reason why this should not run on macOS?

File: Public/Src/Deployment/NugetPackages.dsc [hunk 3]
```diff
@@ -8,17 +8,18 @@ import * as Managed from "Sdk.Managed.Shared";
...
         ? "BuildXL"
         : "Microsoft.BuildXL";
+
     const packageTargetFolder = BuildXLSdk.Flags.isMicrosoftInternal
         ? r`${qualifier.configuration}/pkgs`
         : r`${qualifier.configuration}/public/pkgs`;
 
-
-    const net472 = pack({
+    const net472 = Context.getCurrentHost().os === "macOS" ? undefined : pack({
```
Comment (dannyvv) [inline Public/Src/Deployment/NugetPackages.dsc]: >t.getCurrentHost().os === "macO [](start = 25, length = 31)

perhaps add a local boolean shouldProduceAllNugetPacakges. Reason is that the check here is NOT the inverse of the check on line 113. If we add a third OS we'll end up with a funny split. so this should be always `=== "win"` and `!== "win"` or  `"macOS"`, but not `=== "win"` on one and `=== "macOs"` on the other :D

Comment (Kr1sso) [reply Public/Src/Deployment/NugetPackages.dsc]: Yeah, will refactor this, its a WIP in the PR title 💃

File: Public/Src/Deployment/buildXL.dsc (hunk 2 above)
Comment (Kr1sso) [reply Public/Src/Deployment/buildXL.dsc]: Adding it.

**Pull request closed without merging**